docs/transaction-metrics.asciidoc

```diff
@@ -0,0 +1,93 @@
+[x-pack]
+[[transaction-metrics]]
+== Configure transaction metrics
+
+++++
+<titleabbrev>Transaction metrics</titleabbrev>
+++++
+
+When enabled, {beatname_uc} produces the transaction histogram metrics that are used to power the APM app.
+Shifting this responsibility from APM app to APM Server results in improved query performance.
+
+Example config file:
+
+["source","yaml"]
+----
+apm-server:
+  aggregation:
+    transactions:
+      enabled: true
+      interval: 1m
+  sampling:
+    keep_unsampled: false
+----
+
+
+[float]
+[[configuration-aggregation]]
+=== Configuration options: `apm-server.aggregation.transactions.*`
+
+[[transactions-enabled]]
+[float]
+==== `enabled`
+
+Enables the collection and publishing of transaction metrics.
+This setting improves query performance in the APM app.
+
+Default: `false`.
+
+[[transactions-interval]]
+[float]
+==== `interval`
+
+Controls the frequency of metrics publication.
+
+Default: `1m`.
+
+[[transactions-max_groups]]
+[float]
+==== `max_groups`
+
+Maximum number of transaction groups to keep track of.
+Once exceeded, APM Server devolves into recording a metrics document for each transaction that is not in one
+of the transaction groups being tracked.
+
+Default: `1000`.
+
+[[transactions-hdrhistogram_significant_figures]]
+[float]
+==== `hdrhistogram_significant_figures`
+
+The fixed, worse-case percentage error (specified as a number of significant digits)
+to maintain for recorded metrics.
+Supported values are `1` through `5`.
+See {ref}/search-aggregations-metrics-percentile-aggregation.html#_hdr_histogram[HDR histogram] for more information.
+
+Default: `2`.
+
+[[transactions-lru_size]]
+[float]
+==== `rum.user_agent.lru_size`
+
+This option controls the cache size of RUM user-agent strings.
+
+RUM "page-load" transactions are aggregated on the user-agent name, which requires user-agent parsing.
+To avoid parsing every user-agent, a cache of user-agent strings is maintained.
+
+Default: `5000`.
+
+[float]
+[[configuration-sampling]]
+=== Configuration options: `apm-server.sampling.*`
+
+[[sampling-keep_unsampled]]
+[float]
+==== `keep_unsampled`
```

Comment:
I'm not sure if we should document this one. I think this will end up going away in favour of the updating the agents to stop sending unsampled transactions. Also, I suspect this config might be incompatible with RUM/User Experience monitoring, where we probably want to record individual requests without a full sampled trace.

Reply:
If we do remove this config from the docs, do the `apm-server.aggregation.transactions.*` configs still result in storage reduction? I want to make sure your previous suggested change still holds true if we remove this.

Reply:
Ahh good point. Indeed, storage reduction relies on us discarding the unsampled transaction docs.
I guess we should keep it then, and just mark it as experimental.